JitsiConference.js

```diff
@@ -1362,53 +1362,84 @@ JitsiConference.prototype.onRemoteTrackRemoved = function(removedTrack) {
+        jingleSession,
+        jingleOffer) {
+
+    let rejectReason;
+    const role = this.room.getMemberRole(jingleSession.remoteJid);
+
+    if (role !== 'moderator') {
+        rejectReason = {
+            reason: 'security-error',
+            reasonDescription: 'Only focus can start new sessions',
+            errorMsg: 'Rejecting session-initiate from non-focus and'
+                        + `non-moderator user: ${jingleSession.remoteJid}`
+        };
+    } else if (!RTCBrowserType.isP2PSupported()) {
+        rejectReason = {
+            reason: 'unsupported-applications',
+            reasonDescription: 'P2P not supported',
+            errorMsg: 'This client does not support P2P connections'
+        };
+    } else if (!this.isP2PEnabled() && !this.isP2PTestModeEnabled()) {
+        rejectReason = {
+            reason: 'decline',
+            reasonDescription: 'P2P disabled',
+            errorMsg: 'P2P mode disabled in the configuration'
+        };
+    } else if (this.p2pJingleSession) {
+        // Reject incoming P2P call (already in progress)
+        rejectReason = {
+            reason: 'busy',
+            reasonDescription: 'P2P already in progress',
+            errorMsg: 'Duplicated P2P "session-initiate"'
+        };
+    }
+
+    if (rejectReason) {
+        this._rejectIncomingCall(jingleSession, rejectReason);
+    } else {
+        this._acceptP2PIncomingCall(jingleSession, jingleOffer);
+    }
+};
+
+/**
+ * Handles an incoming call event.
  */
 JitsiConference.prototype.onIncomingCall = function(
         jingleSession,
         jingleOffer,
         now) {
     // Handle incoming P2P call
     if (jingleSession.isP2P) {
-        const role = this.room.getMemberRole(jingleSession.remoteJid);
+        this._onIncomingCallP2P(jingleSession, jingleOffer);
+    } else {
+        if (!this.room.isFocus(jingleSession.remoteJid)) {
+            const description = 'Rejecting session-initiate from non-focus.';
 
-        if (role !== 'moderator') {
-            // Reject incoming P2P call
-            this._rejectIncomingCallNonModerator(jingleSession);
-        } else if (!RTCBrowserType.isP2PSupported()) {
-            // Reject incoming P2P call (already in progress)
-            this._rejectIncomingCall(
-                jingleSession, {
-                    reasonTag: 'unsupported-applications',
-                    reasonMsg: 'P2P not supported',
-                    errorMsg: 'This client does not support P2P connections'
-                });
-        } else if (!this.isP2PEnabled() && !this.isP2PTestModeEnabled()) {
             this._rejectIncomingCall(
                 jingleSession, {
-                    reasonTag: 'decline',
-                    reasonMsg: 'P2P disabled',
-                    errorMsg: 'P2P mode disabled in the configuration'
+                    reason: 'security-error',
+                    reasonDescription: description,
+                    errorMsg: description
```

Comment:
Yes. I added a message since the old code didn't have one at all. But I fail to understand why in the other places we use two different strings, with similar length and similar meanings.